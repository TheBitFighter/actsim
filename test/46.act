defchan e1of1 <: chan(enum<1>) (bool r; bool a)
{
  methods {
    set {
	r+
    }
    send_rest {
    	[a];r-;[~a]
    }
    get {
	[r]
    }
    send_init {
        r-
    }
    recv_rest {
        a+;[~r];a-
    }
    recv_probe = r;
  }
}


defproc src (e1of1 x)
{
  int i;
  chp {
     i := 0;
    *[ i < 100 -> x!; i := i + 1; log("sent!") ]
  }
}

defproc sink (e1of1 x)
{
  prs {
    x.r -> x.a+
   ~x.r -> x.a-
  }
}

defproc test ()
{
   src s;
   sink t(s.x);
}

