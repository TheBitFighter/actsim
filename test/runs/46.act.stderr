WARNING: src<>: substituting chp model (requested prs, not found)
